lib/AST/Module.cpp

```diff
@@ -1996,6 +1996,18 @@ StringRef ModuleDecl::getModuleLoadedFilename() const {
   return StringRef();
 }
 
+bool ModuleDecl::isSDKModule() const {
+  if (getASTContext().SearchPathOpts.getSDKPath().empty())
+    return false;
+
+  auto sdkPath = SmallString<8>(),
+       modulePath = SmallString<8>();
+  llvm::sys::path::native(getASTContext().SearchPathOpts.getSDKPath(), sdkPath);
+  llvm::sys::path::native(getModuleSourceFilename(), modulePath);
+
+  return modulePath.startswith(sdkPath);
```

Comment:
Is there really no `startswith`-equivalent path API? There's an iterator API for going over components, so this surprises me greatly. Another way to do this (assuming the roots are the same) would be:
```
auto si = path::begin(sdkPath), se = path::end(sdkPath)
for (auto mi = path::begin(modulePath), me = path::end(modulePath); si != se && mi != me; ++si, ++mi) {
  if (*si != *mi)
    return false;
}
return si == se;
```

Which would avoid a bunch of string creation (you probably want something more like 64/128 for the default path sizes if you want to keep yours).

Reply:
Thank you for the suggestion, I applied this logic in the latest version.